src/modules/ModerationModule.cpp

```diff
@@ -30,12 +30,18 @@ void ModerationModule::OnMessage(Client& client, const Message& message)
 			{
 				auto authorID = message.author().id();
 				::Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN,
-					[this, message, &client, authorID](bool result)
+					[this, message, &client, authorID, channel](bool result)
 				{
 					if (!result)
 					{
 						client.deleteMessage(message.channelId(), message.id());
-						client.createMessage(message.channelId(), "**<@" + QString::number(authorID) + ">, invitation link of servers aren't allowed in any channels on this server. Please take it to DMs!**");
+						UmikoBot::Instance().createDm(authorID)
+							.then([authorID, &client, message](const Channel& channel)
+								{
+									client.createMessage(channel.id(), "**<@" + QString::number(authorID) + ">, invitation link of servers aren't allowed in any channels on this server. Please take it to DMs!**");
+									client.createMessage(channel.id(), "**Here is your message which you posted in the server:**\n" + message.content());
+								});
+						client.createMessage(channel.id(), "**<@" + QString::number(authorID) + ">, invitation link of servers aren't allowed in any channels on this server. Please take it to DMs!**");
```

Comment:
I don't think we need to send this message. They should already have gotten a DM about it, so the only reason for this public message would be for others to see it, which I see no reason for. It would clutter the advertise channel for example!

Reply:
What if their DM is off, from the server?

Reply:
Like this 
![image](https://user-images.githubusercontent.com/69952230/102012144-d68c0a00-3d72-11eb-8f22-475c0bba9beb.png)

Reply:
In my opinion, if they have opted out of getting DMs from server members, they have opted out of being informed about their participation in the server at a whole.
If someone behaves badly and I try and reach out to them in DMs to try and solve things, I don't wait around for them to accept a friend request. I take action immediately. Not saying anything is valid in this case.